### PR TITLE
WIP: C++ LLVM wrappers for ORCJIT

### PR DESCRIPTION
**Pull request by harlanhaskins**
This won't work for a while, not until SwiftPM gets proper support for C++ dependencies.

### CODE REVIEW

File: Sources/LLVMWrappers/include/LLVMWrappers.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#ifndef LLVMWrappers_h
+#define LLVMWrappers_h
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void *_Nullable LLVMCreateOrcMCJITReplacement(void *_Nonnull module,
```
Comment (CodaFi) [inline Sources/LLVMWrappers/include/LLVMWrappers.h:8]: What happened to this prototype?

Comment (harlanhaskins) [reply Sources/LLVMWrappers/include/LLVMWrappers.h:8]: This is pretty much copy-pasted from Trill. This is just so we can avoid including the LLVM-C headers as well.

Comment (CodaFi) [reply Sources/LLVMWrappers/include/LLVMWrappers.h:8]: I mean, syntactically.

Comment (harlanhaskins) [reply Sources/LLVMWrappers/include/LLVMWrappers.h:8]: I don't see anything wrong. The second param is on the next line and GitHub for some reason isn't  showing the leading whitespace.

**Comment by harlanhaskins** [comment]
Closing this PR because it won't be possible for a long time.

**Pull request closed without merging**